app/components/manageable_authenticator_component.rb

```diff
@@ -0,0 +1,14 @@
+class ManageableAuthenticatorComponent < BaseComponent
+  attr_reader :configuration, :manage_api_url, :manage_url, :tag_options
+
+  def initialize(configuration:, manage_api_url:, manage_url:, **tag_options)
+    if !configuration.respond_to?(:name) || !configuration.respond_to?(:created_at)
+      raise ArgumentError, '`configuration` is invalid, it must respond to `name` and `created_at`'
+    end
+
+    @configuration = configuration
+    @manage_url = manage_url
```

Comment:
> I'm not seeing where `manage_url` gets used? and since it looks like its getting set from the delete URL, should we call it `delete_url` or `remove_url` instead?

Yeah, that's a bit of leftover code that I expected to clean up as part of the implementation of "delete" and/or "reauthentication", depending if we feel we should enforce reauthentication for both renames as well as delete.

What I imagined is:

- If we want reauthentication enforced for both, we'd create a fallback experience similar to what we have for managing a phone (which allows you to change or delete a phone on a separate screen), and this `manage_url` would be that destination
- If we only care about reauthentication for deletion (which is where I'm leaning), we could just keep the existing delete confirmation screen, in which case we'd swap out `manage_url` with something like `delete_url`

Reply:
This also ties into the question of no-JavaScript support. I think I'd imagined if we wanted to support a no-JavaScript fallback, the `manage_url` is where those users would be directed, and we'd probably want to build out an experience like I described in the first bullet point of my previous comment.